dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java

```diff
@@ -215,41 +214,50 @@ private PageView getPageMetadata(final HttpServletRequest request, final HttpSer
             }
         }
 
-        Template template = mode.showLive ? (Template) this.versionableAPI.findLiveVersion(page.getTemplateId(), user, mode.respectAnonPerms) :
+        final Template template = mode.showLive ? (Template) this.versionableAPI.findLiveVersion(page.getTemplateId(), user, mode.respectAnonPerms) :
                 (Template) this.versionableAPI.findWorkingVersion(page.getTemplateId(), user, mode.respectAnonPerms);
 
-        final List<Container> templateContainers = this.templateAPI.getContainersInTemplate
-                (template, user, RESPECT_FE_ROLES);
-        
-        templateContainers.addAll(APILocator.getContainerAPI().getContainersOnPage(page));
+        final TemplateLayout layout = DotTemplateTool.themeLayout(template.getInode());
 
-        
+        final Map<String, ContainerView> mappedContainers = this.getMappedContainers(template, user);
 
-        
-        final Map<String, ContainerView> mappedContainers = new LinkedHashMap<>();
-        for (final Container container : templateContainers) {
-            final List<ContainerStructure> containerStructures = this.containerAPI
-                    .getContainerStructures(container);
-            String rendered = null;
-            if (isRendered) {
-                try {
-                    rendered = VelocityUtil.mergeTemplate("/live/" + container.getIdentifier() +
-                            ".container", velocityContext);
-                } catch (Exception e) {
-                    throw new DotDataException(String.format("Container '%s' could not be " +
-                            "rendered via " + "Velocity.", container.getIdentifier()), e);
-                }
+        if (isRendered) {
+            renderContainer(mappedContainers, velocityContext);
+        }
+
+        return new PageView(site, template, mappedContainers, page, layout);
+    }
+
+    private void renderContainer(final Map<String, ContainerView> containers, final Context velocityContext )
+            throws DotSecurityException, DotDataException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.dotmarketing.exception.DotSecurityException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)